x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.test.ts

```diff
@@ -0,0 +1,456 @@
+        mount(mappingServerProps);
+
+        http.put.mockReturnValue(Promise.resolve(mappingServerProps));
+        RoleMappingsLogic.actions.handleSaveMapping();
+
+        expect(http.put).toHaveBeenCalledWith(`/api/app_search/role_mappings/${asRoleMapping.id}`, {
+          body: JSON.stringify(body),
+        });
+        await nextTick();
+
+        expect(navigateToUrl).toHaveBeenCalled();
+        expect(setSuccessMessage).toHaveBeenCalled();
+      });
+
+      it('sends array when "accessAllEngines" is false', () => {
+        const engine = engines[0];
+
+        mount({
+          ...mappingServerProps,
+          accessAllEngines: false,
+          selectedEngines: new Set([engine.name]),
+        });
+
+        http.put.mockReturnValue(Promise.resolve(mappingServerProps));
+        RoleMappingsLogic.actions.handleSaveMapping();
+
+        expect(http.put).toHaveBeenCalledWith(`/api/app_search/role_mappings/${asRoleMapping.id}`, {
+          body: JSON.stringify({
+            ...body,
+            accessAllEngines: false,
+            engines: [engine.name],
+          }),
+        });
+      });
+
+      it('handles error', async () => {
+        http.post.mockReturnValue(Promise.reject('this is an error'));
+        RoleMappingsLogic.actions.handleSaveMapping();
+        await nextTick();
+
+        expect(flashAPIErrors).toHaveBeenCalledWith('this is an error');
+      });
+    });
+
+    describe('handleDeleteMapping', () => {
+      let confirmSpy: any;
+
+      beforeEach(() => {
+        confirmSpy = jest.spyOn(window, 'confirm');
+        confirmSpy.mockImplementation(jest.fn(() => true));
+      });
+
+      afterEach(() => {
+        confirmSpy.mockRestore();
+      });
+
+      it('returns when no mapping', () => {
+        RoleMappingsLogic.actions.handleDeleteMapping();
+
+        expect(http.delete).not.toHaveBeenCalled();
+      });
+
+      it('calls API and navigates', async () => {
+        mount(mappingServerProps);
+        http.delete.mockReturnValue(Promise.resolve({}));
+        RoleMappingsLogic.actions.handleDeleteMapping();
+
+        expect(http.delete).toHaveBeenCalledWith(
+          `/api/app_search/role_mappings/${asRoleMapping.id}`
+        );
+        await nextTick();
+
+        expect(navigateToUrl).toHaveBeenCalled();
+        expect(setSuccessMessage).toHaveBeenCalled();
+      });
+
+      it('handles error', async () => {
+        mount(mappingServerProps);
+        http.delete.mockReturnValue(Promise.reject('this is an error'));
+        RoleMappingsLogic.actions.handleDeleteMapping();
+        await nextTick();
+
+        expect(flashAPIErrors).toHaveBeenCalledWith('this is an error');
+      });
+
+      it('will do nothing if not confirmed', async () => {
+        mount(mappingServerProps);
+        window.confirm = () => false;
+        RoleMappingsLogic.actions.handleDeleteMapping();
+
+        expect(http.delete).not.toHaveBeenCalled();
+        await nextTick();
+      });
```

Comment:
Actually fwiw I don't think this entire test needs to be async

```suggestion
      it('will do nothing if not confirmed', () => {
        mount(mappingServerProps);
        jest.spyOn(window, 'confirm').mockReturnValueOnce(false);

        RoleMappingsLogic.actions.handleDeleteMapping();

        expect(http.delete).not.toHaveBeenCalled();
      });
```

Reply:
https://github.com/elastic/kibana/pull/94461/commits/59e9d9335d9fecef6e69b1fc13c57e9aae61ff65